src/engine/TextLimit.cpp

```diff
@@ -0,0 +1,162 @@
+    } else if (idTable[i][textRecordColumn_] !=
+               idTable[i - 1][textRecordColumn_]) {
+      // Case: new text record.
+      if (currentEntityCount >= n_) {
+        // Case: new text record and reached the limit.
+        idTable.erase(idTable.begin() + i);
+        --i;
+      } else {
+        // Case: new text record but limit not reached yet.
+        ++currentEntityCount;
+      }
+    }
+    // If the we don't have a new text record we don't have to do anything.
+
+    // NOTE: This results that for each entity there only max n texts not
+    // entries!!! So there can be two entries that share the same entity and
+    // text in the result. Could be confusing if only entity and text are
+    // selected.
+  }
+
+  return {std::move(idTable), resultSortedOn(),
+          childRes->getSharedLocalVocab()};
+}
+
+// _____________________________________________________________________________
+VariableToColumnMap TextLimit::computeVariableToColumnMap() const {
+  return child_->getVariableColumns();
+}
+
+// _____________________________________________________________________________
+size_t TextLimit::getResultWidth() const { return child_->getResultWidth(); }
+
+// _____________________________________________________________________________
+size_t TextLimit::getCostEstimate() {
+  size_t sizeChild = child_->getSizeEstimate();
+  return sizeChild + log2(sizeChild) * sizeChild;
+}
+
+// _____________________________________________________________________________
+uint64_t TextLimit::getSizeEstimateBeforeLimit() {
+  return child_
+      ->getSizeEstimate();  // QUESTION: Why is getSizeEstimateBeforeLimit() not
+                            // possible? And how come we don't have to implement
+                            // getSizeEstimate() here?
+}
+
+// _____________________________________________________________________________
+vector<ColumnIndex> TextLimit::resultSortedOn() const {
+  return {};  // NOTE: sorted on entity, then score, then text all descending.
+              // But apparently this expects an ascending sort. Possible
+              // solution: sort ascending but go through list from back to
+              // front.
+}
+
+// _____________________________________________________________________________
+string TextLimit::getDescriptor() const {
+  std::ostringstream os;
+  os << "TextLimit with limit n: " << n_;
+  return os.str();
+}
+
+// _____________________________________________________________________________
+string TextLimit::getCacheKeyImpl() const {
+  std::ostringstream os;
+  os << "TEXT LIMIT: "
+     << " with n: " << n_ << ", with child: " << child_->getCacheKey()
+     << " and ColumnIndices: " << std::to_string(textRecordColumn_) << ", {";
+  for (const auto& column : entityColumns_) {
+    os << std::to_string(column) << ", ";
+  }
+  os.seekp(-2, os.cur);
+  os << "}, {";
+  for (const auto& column : scoreColumns_) {
+    os << std::to_string(column) << ", ";
```

Comment:
No need for the `to_string` calls, the operator `<<` handles numeric values just fine.